lib/serializer.js

```diff
@@ -157,8 +161,17 @@ function _getSerializedContext(err) {
         _.omit(err, self._knownFields) :
         {};
 
-    if (topLevelFields.domain instanceof domain.Domain) {
-        topLevelFields = _.omit(topLevelFields, [ 'domain' ]);
+    // We don't want to load domains just to check if topLevelFields.domain is
+    // a Domain instance, so first we make sure domains are already loaded.
+    if (EventEmitter.usingDomains) {
+        if (!domain) {
```

Comment:
I think we can load this even more lazily by loading the `domain` module only when processing a `.domain` property on the `topLevelFields` object.

Reply:
We could, but if we get here domains were already loaded somewhere else. Doing it here saves us a hasOwnProperty/in.